src/main/java/net/minestom/server/lock/LockedElement.java

```diff
@@ -0,0 +1,22 @@
+package net.minestom.server.lock;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents an element that have a {@link Acquirable} linked to it.
+ * <p>
+ * Useful if you want to provide an access point to an object without risking to compromise
+ * the thread-safety of your code.
+ */
+public interface LockedElement {
```

Comment:
This should definitely be generic, feels so wrong having it not be..

Reply:
I agree, this was the case at first but inheritance is an isssue